terminus/geometry/circle.py

```diff
@@ -45,45 +45,66 @@ def __eq__(self, other):
     def __hash__(self):
         return hash((self.center, self.radius))
 
+    def radius(self):
+        return self.radius
+
+    def center(self):
+        return self.center
+
     def almost_equal_to(self, other, decimals=5):
         return self.center.almost_equal_to(other.center, decimals) and abs(self.radius - other.radius) < 1e-5
 
-    def intersection(self, other):
-        d = (other.center - self.center).norm()
-        r1 = self.radius
-        r2 = other.radius
-        if d == 0 and r1 == r2:
+    def intersection(self, other, buffer=1e-7, decimals=7):
+        p = self.center
+        q = other.center
+        A = self.radius
+        B = other.radius
```

Comment:
Can we try to use more meaningful variable names (e.g. `my_center` instead of `p` and `my_radius` instead of `q`)?. I find it really hard to follow the code with all those one-letter variables :disappointed: